### PR TITLE
Always use local registry in all CIs

### PR DESCRIPTION
**Pull request by mcasimir**
Always use a local registry to build compass so there is no need to publish packages to the public registry before building release assets and releasing compass.

### CODE REVIEW

**Review by rose-m — APPROVED** [review]
Nice, looks good as far as I understood it 👍 Little unexpected that `startLocalRegistry` also runs `lernaPublish` :D

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -50,11 +50,11 @@ jobs:
...
 
-      - name: Install verdaccio
-        run: npm install -g verdaccio
-
       - name: Install Dependencies
-        run: npm install --force
+        run: npm ci --force
+
+      - name: Bootstrap
+        run: npx lerna run bootstrap --stream
```
Comment (gribnoysup) [inline .github/workflows/build.yaml:57]: I think these two separate steps can be one now if we are changing things here anyway (also we should probably  not use force, there was a weird win issue, but I'd try it and see if it all works without it)

```suggestion
        run: npm run bootstrap-ci
```

File: packages/hadron-build/lib/generate-package-lock.js [hunk 2]
```diff
@@ -1,8 +1,21 @@
...
+    const result = await fn.call(spinner, ...args);
+    spinner.succeed();
+    return result;
+  } catch (e) {
+    if (spinner.isSpinning) {
+      spinner.fail();
+    }
+    throw e;
+  }
+}
```
Comment (gribnoysup) [inline packages/hadron-build/lib/generate-package-lock.js:18]: Now that this script is part of the build lib I think we can just drop all those progress spinners tbh, no need to move them over also (maybe replace then with debug calls to be consistent with other parts of hadron-build)

File: packages/hadron-build/lib/generate-package-lock.js [hunk 3]
```diff
@@ -86,26 +97,14 @@ async function main() {
     }
   );
 
-  await withProgress(
-    `Writing package lock file to ${path.relative(
-      process.cwd(),
-      path.join(workspacePath, 'package-lock.json')
-    )}`,
-    async () => {
-      // https://docs.npmjs.com/cli/v7/configuring-npm/package-lock-json#file-format
```
Comment (gribnoysup) [inline packages/hadron-build/lib/generate-package-lock.js:95]: I think it helps to have a reference to the format so close, maybe we can keep it somehow?

File: packages/hadron-build/lib/local-registry.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  const { stdout } = await execa('npx', [
+    'lerna', 'ls', '--json'
+  ], { cwd: monorepoPath });
+  return JSON.parse(stdout);
+}
+
+async function writeConfigFile(tempDir) {
+  // creates a verdaccio yaml config for the tmp folder
+  // NOTE: in order to disable the uplink with npm for internal packages
+  // we add overrides for each package.
```
Comment (gribnoysup) [inline packages/hadron-build/lib/local-registry.js:44]: Not sure I understand why we need the overrides. `verdaccio` will not publish anything to remote and for installing it doesn't sound too bad to have them or am I missing something?

File: packages/notary-service-client/.depcheckrc [hunk 5]
```diff
@@ -1,3 +1 @@
-ignores: [
-  "execa"
-]
+ignores: []
```
Comment (gribnoysup) [inline packages/notary-service-client/.depcheckrc:1]: We can probably drop the whole file if it's empty, right?

**Review by gribnoysup — APPROVED** [review]
Great stuff! Left a few comments in places that caught my eye, but nothing important

**Comment by gribnoysup** [comment]
> Little unexpected that startLocalRegistry also runs lernaPublish

Yeah, I think Michael has a good point here, maybe we can change it to `startRegistryAndPublish` or something?

File: .github/workflows/build.yaml (hunk 1 above)
Comment (mcasimir) [reply .github/workflows/build.yaml:57]: Is it ok to don't have `--force`?

File: packages/hadron-build/lib/generate-package-lock.js (hunk 2 above)
Comment (mcasimir) [reply packages/hadron-build/lib/generate-package-lock.js:18]: sounds good

File: packages/hadron-build/lib/generate-package-lock.js (hunk 3 above)
Comment (mcasimir) [reply packages/hadron-build/lib/generate-package-lock.js:95]: ah i don't think i removed this on purpose, will get it back!

File: .github/workflows/build.yaml (hunk 1 above)
Comment (gribnoysup) [reply .github/workflows/build.yaml:57]: I think so, I think we were running into a weird windows issue that is gone now, but you'll have to try 😄

File: packages/hadron-build/lib/local-registry.js (hunk 4 above)
Comment (mcasimir) [reply packages/hadron-build/lib/local-registry.js:44]: yeah .. here the overrides serves a different purpose. With that setup we can publish the packages with same version (without bumping anything first), verdaccio won't even look for those on npm so `lerna publish from-package` won't detect them as already published.

**Comment by mcasimir** [comment]
> > Little unexpected that startLocalRegistry also runs lernaPublish
> 
> Yeah, I think Michael has a good point here, maybe we can change it to `startRegistryAndPublish` or something?

Makes a lot of sense

File: packages/hadron-build/lib/local-registry.js (hunk 4 above)
Comment (gribnoysup) [reply packages/hadron-build/lib/local-registry.js:44]: Ahh, it all makes sense now, thanks!

File: .github/workflows/build.yaml [hunk 6]
```diff
@@ -50,11 +50,8 @@ jobs:
         # Can be changed back to npm@7 when https://github.com/npm/cli/issues/3637 is resolved
         run: npm install -g npm@7.20.2
 
-      - name: Install verdaccio
-        run: npm install -g verdaccio
-
       - name: Install Dependencies
-        run: npm install --force
+        run: npm run bootstrap-ci
```
Comment (gribnoysup) [inline .github/workflows/build.yaml:54]: Sorry, I think I did a wrong suggest here, we don't have `bootstrap-ci` script, only `bootstrap-evergreen` (that we probably should rename) 🙈

**Pull request merged**